# Related Work #

**jBio**: is a JavaScript library proposed by the Rost Lab at Technische Universität München. Their proposal is underlined by the use of methodologies such as object-oriented programming with inheritance and polymorphism, asynchronous JavaScript + XML (AJAX), and jQuery, a leading JavaScript library that facilitates rapid development. Hence, by using jBio developers can rapidly design and deploy true web 2.0 applications. https://www.rostlab.org/owiki/index.php/JBio

**bio-js**: This is an attempt at a bioinformatics framework in JavaScript. Uses the existing framework PrototypeJS (http://www.prototypejs.org/) to facilitate the process and to ensure that the project is compatible with most operating systems and browsers. They are using inheritance encapsulated in a unique variable called Bio. This project is under developing. The classes already implemented is mainly to manage the DNA sequence element. http://code.google.com/p/bio-js/

**BioViews**: is a Java-Based tools for genomic data visualization. Basically, is a prototype genome browser applet (BioViews) that incorporates a three-level graphical view of genomic data: a physical map, an annotated sequence map, and a DNA sequence display (see image below). This is a single application consists of two components, a map widget for displaying features along a linear map (which is used for both the physical map and annotated sequence views) and a DNA display widget for showing nucleotide sequences at base-pair resolution

![http://www.ncbi.nlm.nih.gov/pmc/articles/PMC310693/bin/gr.1f3.jpg](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC310693/bin/gr.1f3.jpg)

DNA view.
Source: http://www.ncbi.nlm.nih.gov/pmc/articles/PMC310693/

**Hypergeny and GenomeView**: these two softwares was found as part of a extensive list of software developed by the center of excellence Bioinformatics and Systems Biology at Ghent University. It consists in Java-based applications enabled to integrated it with third party web pages using java applet. By one hand, GenomeView is a stand-alone sequence browser specifically designed to visualize and manipulate a multitude of genomics data interactively. By the other hand, Hypergeny allows hyperbolic visualisation of very large tree structures (phylogenetic trees).  http://bioinformatics.psb.ugent.be/software

**bioWidgets**: is a toolkit collection of components based on Java Beans(tm) used for development of graphics applications and/or applets in the genomics domain. These components can be orchestrated in order to create your own application. It was developed by the Computational Biology and Informatics Laboratory at the University of Pennsylvania. The documentation introduce three applications developed with the concept of bioWidget toolkit: AnnotView, BlastView and AlignView. However, an example of functional application was not found. http://www.cbil.upenn.edu/bioWidgets/

**Cytoscape**: is a very complete stand alone application for visualizing molecular interaction networks and biological pathways and integrating these networks with annotations, gene expression profiles and other state data. Based on Java™ technology. It can be extended by implementing plug-ins using Cytoscape open API.

![http://www.cytoscape.org/images/intro_14.png](http://www.cytoscape.org/images/intro_14.png)

A screenshot of Cytoscape
source: http://www.cytoscape.org/what_is_cytoscape.html

**Other**: there are some examples of fully functional biological components at http://www.ebi.ac.uk/~jvillave/BioWidgets/samples/samples.html for DNA sequences and ontology trees. It was developed using plug-ins in jQuery. The problem here is the lack of documentation in order to reuse or extend the code.